plugins/nf-co2footprint/src/main/nextflow/co2footprint/CO2FootprintFactory.groovy

```diff
@@ -173,7 +173,7 @@ class CO2FootprintFactory implements TraceObserverFactory {
          * Remaining factors
          */
         // PUE: efficiency coefficient of the data centre
-        def pue = 1.67
+        def pue = config.getPUE()
```

Comment:
Why do you think it is better, is there any convention in nextflow that it should always be added? 

I thought it is only added in order to differentiate variables with the same name, such as when assigning the argument `session.config` to `this.config` [here](https://github.com/qbic-pipelines/nf-co2footprint/blob/70f67a2db59951a89d364241fa90c07d79dc5318/plugins/nf-co2footprint/src/main/nextflow/co2footprint/CO2FootprintFactory.groovy#L76), but not when there is no ambiguity, as [here](https://github.com/nextflow-io/nextflow/blob/fd4ff04f827ebca0f25efd3b079a68f390a82079/modules/nextflow/src/main/groovy/nextflow/trace/DefaultObserverFactory.groovy#L42) in Nextflow for example.

Reply:
Mainly for consistency, we use it for example with `getFile()` or `getSummaryFile()` https://github.com/qbic-pipelines/nf-co2footprint/blob/bc7863fd1b72f3aafee8533775ca870777788768/plugins/nf-co2footprint/src/main/nextflow/co2footprint/CO2FootprintFactory.groovy#L80
and because I find it more clear. But there isn't a Nextflow convention, looks also good if you want to leave it as it is now

Reply:
ah yes, that's because there is a `session.config` as well in the methods scope